### ISSUE TITLE
Support to custom quota project

### ISSUE THREAD

**Issue by ruyadorno** [GoogleCloudPlatform/cloud-sql-nodejs-connector#62]
## Feature Description

Add param to allow an user to define a custom quota project when connecting to CloudSQL Admin APIs.

**Comment by edosrecki** [comment]
@ruyadorno this is already supported, by using `GOOGLE_CLOUD_QUOTA_PROJECT` env var:

https://github.com/googleapis/google-auth-library-nodejs/blob/5f296c37390ba71aa3aa1e4259ff7c525eeaad7c/src/auth/googleauth.ts#L339

**Comment by enocom** [comment]
Nice find! We should document that here, but also add an explicit API option too to match the other Connectors.

**Comment by edosrecki** [comment]
If I did not miss something, there is currently no other way to pass `quotaProjectId` to [google-auth-library](https://www.npmjs.com/package/google-auth-library). It either reads it from the credentials JSON, or overrides it from the `GOOGLE_CLOUD_QUOTA_PROJECT` env var.

**Comment by ruyadorno** [comment]
@edosrecki you're right it doesn't look like there's a programmatic API support to it other than the environment variable.

I believe that there's a case to be made to contributing a new API to **google-auth-library** to add that support.

**Comment by enocom** [comment]
In that case, we might let this sit for now. I'd suggest folks upvote the original comment if they want this and we'll prioritize accordingly.

**Comment by ruyadorno** [comment]
Caught up with @danielbankhead last week that showed me https://github.com/googleapis/google-auth-library-nodejs/pull/1663 which normalized a `quotaProjectId` option for all auth clients. I believe that with that shipped in `google-auth-library@9.2.0` we should be unblocked now to add programmatic API support to it.

**Comment by ruyadorno** [comment]
Given that https://github.com/GoogleCloudPlatform/cloud-sql-nodejs-connector/pull/238 enabled using any custom auth to be set I believe that a reliable way to support this is to document setting the value via `auth` property, e.g:

```js
import {GoogleAuth} from 'google-auth-library';
import {Connector} from '@google-cloud/cloud-sql-connector';

const connector = new Connector({
  auth: new GoogleAuth({
    quotaProjectId: "<custom quota project>",
  }),
});
```

Any thoughts @danielbankhead ?
